packages/rehype-shiki/src/index.mjs

```diff
@@ -17,6 +19,21 @@ import { createHighlighter } from './highlighter.mjs';
 
 const { shiki, getLanguageDisplayName, highlightToHast, highlightToHtml } =
   createHighlighter({
+    // We use the faster WASM engine on the server instead of the web-optimized version.
+    //
+    // Currently we fall back to the JavaScript RegEx engine
+    // on Cloudflare workers because `shiki/wasm` requires loading via
+    // `WebAssembly.instantiate` with custom imports, which Cloudflare doesn't support
+    // for security reasons.
+    //
+    // TODO(@avivkeller): When available, use `OPEN_NEXT_CLOUDFLARE` environmen
+    // variable for detection instead of current method, which will enable better
+    // tree-shaking.
+    // Reference: https://github.com/nodejs/nodejs.org/pull/7896#issuecomment-3009480615
+    engine:
+      'Cloudflare' in globalThis
```

Comment:
Is this the best way of verifying this? But guess it makes sense

Reply:
No, see the TODO comment above